src/helpers/blockTime.js

```diff
@@ -0,0 +1,29 @@
+const MILLISECONDS_IN_A_SECOND = 1000
+
+export const blockTime = (eth) =>
+  async (blockNumber, showBlock = true, averageBlockTime = 13.965) => {
+    let timestamp
+    const currentBlock = await eth.getBlockNumber()
+
+    if (currentBlock >= blockNumber) {
+      timestamp = (await eth.getBlock(blockNumber)).timestamp * MILLISECONDS_IN_A_SECOND
+    } else {
+      const { timestamp: currentTimestamp } = await eth.getBlock(currentBlock)
+      const blockDuration = (blockNumber - currentBlock) * averageBlockTime
+      timestamp = (currentTimestamp + blockDuration) * MILLISECONDS_IN_A_SECOND
+    }
+
+    return {
+      type: 'string',
+      value: `${new Date(timestamp).toDateString()}${showBlock ? ` (block number: ${blockNumber})` : ''}`
```

Comment:
We already use `date-fns` in this library (see `helpers/transformTime) and we should:

- Provide a default format
- Allow users to select their own format

Moreover, for calculating future time, we may want to add an indication that the time is an estimate.